gusto/io.py

```diff
@@ -15,6 +15,10 @@
 __all__ = ["pick_up_mesh", "IO"]
 
 
+class GustoIOError(IOError):
```

Comment:
OK this is fun! Should we use this in place of our other `IOError`?

Reply:
Which is on line 315...

Reply:
Done!

Yes, this is a neat trick if you want to catch your own IO exceptions, but not general IO exceptions.

For instance if the disk is full you probably want to let that error get raised as an `IOError` and crash the program, but if it's a known Gusto shortcoming that you may wish to handle (like `PointDataOutput` does not work in parallel) you can raise a `GustoIOError` and catch & handle the exception.

If you like the pattern, it might be worth having an `exceptions.py` in the repo with all of your custom exceptions, but I didn't think it was worth it for 2 (there's also a `LoggingError` which I should have probably called `GustoLoggingError`).